context/app/static/js/components/detailPage/ProcessedData/ProcessedDataWorkspaceMenu.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import { useEventCallback } from '@mui/material/utils';
+import Menu from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
+import AddRounded from '@mui/icons-material/AddRounded';
+
+import SelectableTableProvider from 'js/shared-styles/tables/SelectableTableProvider';
+import { WorkspacesIcon } from 'js/shared-styles/icons';
+
+import { useOpenDialog } from '../../workspaces/WorkspacesDropdownMenu/WorkspacesDropdownMenu';
+import { useCreateWorkspaceForm } from '../../workspaces/NewWorkspaceDialog/useCreateWorkspaceForm';
+import NewWorkspaceDialog from '../../workspaces/NewWorkspaceDialog/NewWorkspaceDialog';
+import AddDatasetsFromSearchDialog from '../../workspaces/AddDatasetsFromSearchDialog/AddDatasetsFromSearchDialog';
+import { useAppContext, useFlaskDataContext } from '../../Contexts';
+import { useTrackEntityPageEvent } from '../useTrackEntityPageEvent';
+
+interface ProcessedDataWorkspaceMenuProps {
+  button: React.ReactNode;
+}
+
+function ProcessedDataWorkspaceMenu({ button }: ProcessedDataWorkspaceMenuProps) {
+  const {
+    entity: { mapped_data_access_level, hubmap_id, uuid, entity_type, status },
+  } = useFlaskDataContext();
+  const isDataset = entity_type === 'Dataset';
+  const { isWorkspacesUser } = useAppContext();
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const open = Boolean(anchorEl);
+  const track = useTrackEntityPageEvent();
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    track({
+      action: 'Open Workspace Menu',
+      label: hubmap_id,
+    });
+    setAnchorEl(event.currentTarget);
+  };
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const {
+    control,
+    errors,
+    removeDatasets,
+    setDialogIsOpen: setOpenCreateWorkspace,
+    dialogIsOpen: createWorkspaceIsOpen,
+    ...rest
+  } = useCreateWorkspaceForm({
+    defaultName: hubmap_id,
+    initialSelectedDatasets: [uuid] ?? [],
+  });
+
+  const openEditWorkspaceDialog = useOpenDialog('ADD_DATASETS_FROM_SEARCH');
+
+  const trackCreateWorkspace = useEventCallback(() => {
+    track({
+      action: 'Start Creating Workspace',
+      label: hubmap_id,
+    });
+    setOpenCreateWorkspace(true);
+    handleClose();
+  });
+
+  const trackAddToWorkspace = useEventCallback(() => {
+    track({
+      action: 'Start Adding Dataset to Existing Workspace',
+      label: hubmap_id,
+    });
+    openEditWorkspaceDialog();
+    handleClose();
+  });
+
+  // Clone the button element and add the onClick handler
+  const buttonWithClickHandler = React.cloneElement(button as React.ReactElement, {
+    onClick: handleClick,
+    'aria-controls': open ? 'basic-menu' : undefined,
+    'aria-haspopup': 'true',
+    'aria-expanded': open ? 'true' : undefined,
+  });
```

Comment:
Why can't we just pass these as props to a mui `Button` component?

Reply:
Because the cases for this component involve very differently styled menu buttons, I opted to let that button be a fully customizable React node prop (screen caps below). This cloned element was an effort to add as many shared properties to that customized button as possible and avoid duplication.

![Screenshot 2024-09-06 at 10 30 31 AM](https://github.com/user-attachments/assets/26898442-db02-44d9-8079-6d159bb251d0)
![Screenshot 2024-09-06 at 10 30 42 AM](https://github.com/user-attachments/assets/40786fa5-e21e-43c0-ada7-84e0c1ea15a2)